waku/waku_archive/retention_policy/retention_policy_size.nim

```diff
@@ -41,29 +41,49 @@ proc init*(T: type SizeRetentionPolicy, size=DefaultRetentionSize): T =
 method execute*(p: SizeRetentionPolicy,
                 driver: ArchiveDriver):
                 Future[RetentionPolicyResult[void]] {.async.} =
+
+  # In SQLite vacuuming is done manually,
+  let dbEngine = driver.getDbType()
+  if dbEngine == "sqlite":
+    return ok()
+
   ## when db size overshoots the database limit, shread 20% of outdated messages 
   # get size of database
-  let dbSize = (await driver.getDatabaseSize()).valueOr:
+  var dbSize = (await driver.getDatabaseSize()).valueOr:
     return err("failed to get database size: " & $error)
 
   # database size in bytes
-  let totalSizeOfDB: int64 = int64(dbSize)
+  var totalSizeOfDB: int64 = int64(dbSize)
+  let retryLimit = 2
+  var retryCounter:int = 0
 
   if totalSizeOfDB < p.sizeLimit:
     return ok()
+  
+    # NOTE: Using SQLite vacuuming is done manually, we delete a percentage of rows
+    # if vacumming is done automatically then we aim to check DB size periodially for efficient
+    # retention policy implementation.
+  while (totalSizeOfDB > p.sizeLimit) and (retryLimit > retryCounter):
```

Comment:
This check, `while (totalSizeOfDB > p.sizeLimit)`, will always be `true` after removing and applying `VACUUM`. We need to think about another approach.

Reply:
moved the while logic to test case and not the core retention policy.